.github/workflows/push-main-deploy-github-pages.yml

```diff
@@ -25,37 +25,42 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  # Build job
-  build:
+
+  build_static_jekyll_site_artifacts:
     runs-on: ubuntu-latest
     steps:
+
       - name: Checkout
         uses: actions/checkout@v3
+
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: '3.1' # Not needed with a .ruby-version file
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
           cache-version: 0 # Increment this number if you need to re-download cached gems
-      - name: Setup Pages
+
+      - name: Enable GitHub Pages and extract various metadata
```

Comment:
It's still unclear to me what this `actions/configure-pages@v3` does exactly.
But I've extracted the "Enable GitHub Pages and extract various metadata" verbiage from the [Action's README](https://github.com/actions/configure-pages#:~:text=GitHub%20Action%20to-,enable%20Pages%20and%20extract%20various%20metadata,-about%20a%20site).